### PR TITLE
Get_user_order performance next trial

### PR DESCRIPTION
**Pull request by josojo**
When we deployed this PR: https://github.com/cowprotocol/services/pull/701 to production, it caused severe performance decreases. This PR reverts the PR + improves the performance with another trick.

 It re-creates the old index that was deleted [here](https://github.com/cowprotocol/services/pull/610/files#diff-2794b4d292d74cb53a35285b59b1c05669ee45a9298edfe40098c720eb8f8433R2) and thereby optimizes the first search query of the union all. This helps especially, by limiting the requested rows in the first query by (OFFSET * OVER_ALL_LIMIT). Especially for small offsets, the query gets its old performance back.

 But let's also discuss in this PR other methods to make the query faster, or whether we should delete old orders. I think with limit orders approaching, the db performance will become more problematic over time.

---

**Testing the previous issue:**
Query that caused problems before can be tested with this query (problematic address mentioned [here](https://cowservices.slack.com/archives/C0361CDD1FZ/p1667035528153689) is already in it). Query needs to be run against xdai readonly db:
https://gist.github.com/josojo/fb73fdfdec6734472fa48d9344d9ad0c
My execution time without the index created in this PR is: 
=>01.081 secs

---

**Why the issue was not caught in the previous PR:** 
I was testing with a too small database. With the size I could already see the difference clearly between the UNION and OR statements, but I could not see the performance degradation of OR. THis shows only later with more orders in the database.

New performance results of `postgres_user_orders_performance_many_users_with_some_orders`

On main branch with `I in 0..1000u32` and `J in 0..300u32`
=> Time per execution 404.366335ms
On this branch with `I in 0..1000u32` and `J in 0..300u32`
=> Time per execution 1.51164ms
On branch: v2.80.2(hotfix in prod) with `I in 0..1000u32` and `J in 0..300u32`
=> Time per execution 2.073314ms

New performance results of `postgres_user_orders_performance_user_with_many_orders`

On branch v2.80.2(hotfix in prod) with `j in 0..100_000u32` 
=>Time per execution 929.626571ms
On this branch:
=> Time per execution 4.141594ms 

But for sure, the performance will degrate with higher Offsets again. :( I think there is nothing I can do about it. Hope you guys have other ideas.

### CODE REVIEW

File: crates/database/src/orders.rs [hunk 1]
```diff
@@ -494,10 +494,16 @@ pub fn user_orders<'a>(
...
 " FROM ", ORDERS_FROM,
 " LEFT OUTER JOIN onchain_placed_orders onchain_o on onchain_o.uid = o.uid",
-" WHERE o.owner = $1 OR onchain_o.sender = $1 ",
+" WHERE o.owner = $1",
+" ORDER BY creation_timestamp DESC LIMIT $2 * ($3 + 1) ) ",
+" UNION ",
+" (SElECT ", ORDERS_SELECT,
+" FROM ", ORDERS_FROM,
+" LEFT OUTER JOIN onchain_placed_orders onchain_o on onchain_o.uid = o.uid",
+" WHERE onchain_o.sender = $1) ",
```
Comment (vkgnosis) [inline crates/database/src/orders.rs:506]: Couldn't this query also get the ORDER BY and LIMIT that you used in the other sub-query?

**Comment by nlordell** [comment]
@josojo - would it be possible to understand what PostgreSQL is doing with these queries using `EXPLAIN ANALYZE` on the query?

**Comment by josojo** [comment]
@nlordell I am not sure what you wanna see from there.

I put the output here
https://docs.google.com/spreadsheets/d/1U15MfFvIQbKmnNP2KA9FB-XuEkbki_IfC4O_K9G7i1o/edit#gid=1863284260
And I compare the difference with and without limit in the subqueries. I highlighted the difference in green. But I am not sure whether this gives me more confidence.

I will also create one more benchmark between this PR and prod code with the benchmark tools.

But I did not make any progress with [duplicating the db](https://cowservices.slack.com/archives/C6Z2XNL5Q/p1667989826720819). Anuj support would be needed for that.

**Review by nlordell — APPROVED** [review]
The approach seems good to me.

**Review by MartinquaXD — APPROVED** [review]

**Pull request merged**